src/cli/java/org/commcare/util/cli/ApplicationHost.java

```diff
@@ -109,15 +112,63 @@ public void setRestoreToDemoUser() {
         mRestoreStrategySet = true;
     }
 
-    public void run() {
+    public void advanceSessionWithEndpoint(String endpointId, String[] endpointArgs) {
+        if (endpointId == null) {
+            return;
+        }
+
+        Endpoint endpoint = mPlatform.getEndpoint(endpointId);
+        if (endpoint == null) {
+            throw new RuntimeException(endpointId + " not found");
+        }
+        if (endpointArgs == null) {
+            endpointArgs = new String[0];
+        }
+
+        mSession.clearAllState();
+        mSession.clearVolatiles();
+
+        EvaluationContext evalContext = mSession.getEvaluationContext();
+        try {
+            Endpoint.populateEndpointArgumentsToEvaluationContext(endpoint, new ArrayList<String>(Arrays.asList(endpointArgs)), evalContext);
+        } catch (Endpoint.InvalidEndpointArgumentsException e) {

```

Comment:
@shubham1g5 Are we really going to have a 100-char limit? Even in python we have 118, and Java is more verbose.

Reply:
I am happy to change it to 120. 100 is a standard at Google where I took the styles files from.

Reply:
It'll affect you more than me, but yeah, bumping it up a little sounds good 🙂